esphome/components/text/text_traits.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <utility>
+
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace text {
+
+enum TextMode : uint8_t {
+  TEXT_MODE_AUTO = 0,
+  TEXT_MODE_TEXT = 1,
+  TEXT_MODE_PASSWORD = 2,
+};
+
+class TextTraits {
+ public:
+  // Set/get the number value boundaries.
+  void set_min_length(int min_length) { min_length_ = min_length; }
```

Comment:
Please use `this->` when referring to member variables
```suggestion
  void set_min_length(int min_length) { this->min_length_ = min_length; }
```